### PR TITLE
Add FDF support to mcpdf

### PR DESCRIPTION
**Pull request by alokdhir**
See title; that's pretty much it.

### CODE REVIEW

**Comment by vog** [comment]
Dear  alokdhir,

Thanks for the contribution! I'd love to see this in Mcpdf. Unfortunately, there are some issues which prevent me from taking your pull-request as-is:

* Most commits have no clear authorship. The author is "root" instead of your name and email address.
* The 4 commits are somewhat chaotic, starting with a merge and other strange stuff. This makes it hard to follow. I recommend to use "git rebase" or "git rebase -i" to clean this up. If this is too much trouble, you could alternatively provide your changes as a single commit that contains everything. (A single commit is perfectly okay here, because the amount of code is not too large, and because all changes are related to a single topic.)
* Within your code, your text editor inserted a mix of tabs and spaces. This makes the diffs harder to read. Please use only spaces for indentation, and don't use tabs at all.
* The command `fill_form_fdf` does not exist in [PDFtk](https://www.pdflabs.com/docs/pdftk-man-page/), so Mcpdf would no longer be a drop-in replacement for PDFtk. Instead, PDFtk has only a single command `fill_form` that accepts both variants. If there is no recognition code for that in the iTextPdf library, this could be implemented as follows:
 * Use a `BufferedInputStream` with `mark(4)` and `reset()`.
 * Look ahead the first 4 bytes of the file.
 * If the first 4 bytes are "`%FDF`", use `FdfReader`, otherwise use `XfdfReader`.

Do you think this is possible for you to fix, or do you need assistance from me?

**Comment by alokdhir** [comment]
Sure - no problem. My primary issue with my approach was your fourth bullet. Your approach to fixing is obviously much better than my hack. I’m not seeing a utility method to recognize fdf vs xfdf.

Thanks!
